src/python_testing/TC_WASHERCTRL_2_1.py

```diff
@@ -0,0 +1,125 @@
+#
+#    Copyright (c) 2024 Project CHIP Authors
+#    All rights reserved.
+#
+#    Licensed under the Apache License, Version 2.0 (the "License");
+#    you may not use this file except in compliance with the License.
+#    You may obtain a copy of the License at
+#
+#        http://www.apache.org/licenses/LICENSE-2.0
+#
+#    Unless required by applicable law or agreed to in writing, software
+#    distributed under the License is distributed on an "AS IS" BASIS,
+#    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#    See the License for the specific language governing permissions and
+#    limitations under the License.
+#
+
+# See https://github.com/project-chip/connectedhomeip/blob/master/docs/testing/python.md#defining-the-ci-test-arguments
+# for details about the block below.
+#
+# === BEGIN CI TEST ARGUMENTS ===
+# test-runner-runs:
+#   run1:
+#     app: ${ALL_CLUSTERS_APP}
+#     app-args: --discriminator 1234 --KVS kvs1 --trace-to json:${TRACE_APP}.json
+#     script-args: >
+#       --storage-path admin_storage.json
+#       --commissioning-method on-network
+#       --discriminator 1234
+#       --passcode 20202021
+#       --PICS src/app/tests/suites/certification/ci-pics-values
+#       --trace-to json:${TRACE_TEST_JSON}.json
+#       --trace-to perfetto:${TRACE_TEST_PERFETTO}.perfetto
+#     factory-reset: true
+#     quiet: true
+# === END CI TEST ARGUMENTS ===
+
+import logging
+import random
+
+import chip.clusters as Clusters
+from chip.interaction_model import Status
+from chip.testing.matter_testing import MatterBaseTest, TestStep, async_test_body, default_matter_test_main
+from mobly import asserts
+
+logger = logging.getLogger(__name__)
+
+
+MAX_SPIN_SPEEDS = 16
+
+
+class TC_WASHERCTRL_2_1(MatterBaseTest):
+
+    def desc_TC_WASHERCTRL_2_1(self) -> str:
+        """Returns a description of this test"""
+        return "[TC-WASHERCTRL-2.1] Optional Spin attributes with DUT as Server"
+
+    def pics_TC_WASHERCTRL_2_1(self) -> list[str]:
+        pics = [
+            "WASHERCTRL.S.F00",
+        ]
+        return pics
+
+    def steps_TC_WASHERCTRL_2_1(self) -> list[TestStep]:
+        steps = [
+            TestStep(1, "Commissioning, already done",
+                     is_commissioning=True),
+            TestStep(2, "TH reads from the DUT the SpinSpeeds attribute"),
+            TestStep(3, "TH reads from the DUT the SpinSpeedCurrent attribute"),
+            TestStep(4, "TH writes a supported SpinSpeedCurrent attribute that is a valid index into the list"
+                        + "of spin speeds (0 to numSpinSpeeds - 1)"),
+            TestStep(5, "After a few seconds, TH reads from the DUT the SpinSpeedCurrent attribute"),
+            TestStep(6, "TH writes an unsupported SpinSpeedCurrent attribute that is other than 0 to DUT")
+        ]
+
+        return steps
+
+    @async_test_body
```

Comment:
Please see comment here - would want to do the same thing in this PR, assuming the lined PR lands before this gets undrafted. https://github.com/project-chip/connectedhomeip/pull/36703#discussion_r1870374327